### PR TITLE
Bug 1786314: OpenStack: support self-signed certificates in the installer

### PR DESCRIPTION
**Pull request by Fedosin**
Recently TLS support was implemented in gophercloud/utils[1], so, by updating the library we allow the installer to work with self-signed certificates without any code modifications.

[1] https://github.com/gophercloud/utils/issues/131

/label platform/openstack

### CODE REVIEW

**Comment by Fedosin** [comment]
`curl: (6) Could not resolve host: github.com` :(

File: pkg/utils/openstack/OWNERS [hunk 1]
```diff
@@ -0,0 +1,7 @@
+# See the OWNERS docs: https://git.k8s.io/community/contributors/guide/owners.md
```
Comment (abhinavdahiya) [inline pkg/utils/openstack/OWNERS:1]: Please no utils package. This package has a very high chance of becoming collections of various different types of packages and causes long term confusion. Please try and use something that is more specific that doesn't have `utils` as directory in path.

File: pkg/auth/openstack/serviceclient.go [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+	"io/ioutil"
+	"net/http"
+
+	"github.com/gophercloud/gophercloud"
+	"github.com/gophercloud/utils/env"
+	"github.com/gophercloud/utils/openstack/clientconfig"
+	"github.com/pkg/errors"
+)
+
+// NewServiceClient is a wrapper function around gophercloud/utils that enables support of self-sighed
```
Comment (staebler) [inline pkg/auth/openstack/serviceclient.go:16]:
```suggestion
// NewServiceClient is a wrapper function around gophercloud/utils that enables support of self-signed
```

**Comment by mandre** [comment]
/hold
We're exploring another way of doing this by adding the support for cacert directly in gophercloud.
https://github.com/gophercloud/utils/pull/131

**Review by staebler — COMMENTED** [review]
/approve

**Comment by EmilienM** [comment]
I think we need to have https://github.com/openshift/installer/pull/4457 merged first, then rebase this one and it'll be good.

**Comment by EmilienM** [comment]
https://github.com/openshift/installer/pull/4457 has landed, please rebase on top of master and this should be ready to go.

**Comment by Fedosin** [comment]
This change is included in https://github.com/openshift/installer/pull/4457

**Pull request closed without merging**